01_fib/hsy3418/main.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func fib(n int) {
+	first := 1
+	second := 1
+	for i := 0; i < n; i++ {
+		switch i {
```

Comment:
consider whether you need this switch at all. If for each case you just do:
```
fmt.Fprintln(out, first)
first, second = second, first+second
```
you will get the same result. The code would then be a lot simpler.

Reply:
Done. Excellent suggestion!